plugins/services/src/js/types/ServiceGroup.ts

```diff
@@ -7,8 +7,21 @@ export type QuotaResources = {
   consumed?: number;
 };
 
+export type ServiceGroupQuotaRoles = {
+  count: number;
+  groupRoleCount: number;
+};
+
+export type ServiceGroupQuotaLimit =
+  | "Enforced"
+  | "Not Enforced"
+  | "Partially Enforced"
+  | "N/A";
+
 export type ServiceGroupQuota = {
   enforced: boolean;
+  limitStatus: ServiceGroupQuotaLimit;
+  serviceRoles?: ServiceGroupQuotaRoles;
```

Comment:
could we provide a default value for `ServiceGroup.quota`, so we could change the type like so?

```diff
export type ServiceGroup = {
  id: string;
  name: string;
-  quota?: null | ServiceGroupQuota;
+  quota: ServiceGroupQuota;
};
```

It looks like we could remove some uncertainty in the code using this type as a follow up.

Reply:
I don't think so, since this type is the return value from graphql queries we want quota to be optional so that you can make queries where you dont' get "quota" and don't make the additional API calls. Granted we're only using this for quota currently, but I think for graphql query types we should stick with flexibility IMO. 

We might be able to remove the null value though. I will look into that when I work on refactoring DL stuff for DCOS-56352